cfg_mgmt/btp_application_certificate.py

```diff
@@ -0,0 +1,256 @@
+        if idx != -1:
+            cert_pem = cert_pem[:idx]
+        data = {
+            'description': desc,
+            'scopes': scopes,
+            'base64': cert_pem,
+        }
+        with self._session_with_cert() as session:
+            resp = session.put(self.url, json=data)
+        resp.raise_for_status()
+        id = resp.json()['certificateId']
+        logger.info(f'Added certificate {id}')
+        return id
+
+    def delete_certificate(self, common_name: str, cert_id: str):
+        data = {
+            'certificateId': cert_id,
+        }
+        with self._session_with_cert() as session:
+            resp = session.delete(self.url, json=data)
+        resp.raise_for_status()
+        logger.info(f'Deleted certificate {common_name} ({cert_id})')
+
+    @staticmethod
+    def _find_cn(dn: str) -> str:
+        for part in dn.split(','):
+            (key,value) = part.split('=')
+            if key == 'CN':
+                return value
+        return None
+
+    def list_certificates_by_base(self, common_name_base: str) -> list[CertInfo]:
+        with self._session_with_cert() as session:
+            resp = session.get(self.clienturl)
+        resp.raise_for_status()
+        certs = []
+        for item in resp.json():
+            id = item['dnId']
+            dn = item['dn']
+            cn = GBaasAppClient._find_cn(dn)
+            if cn:
+                serial_no, base = BtpApplicationCertificate.get_serial_no_from_common_name(cn)
+                if serial_no > 0 and common_name_base == base:
+                    certs.append(CertInfo(id=id, dn=dn, cn=cn, serial_no=serial_no))
+        return certs
+
+
+_str_to_names = {
+    'C': NameOID.COUNTRY_NAME,
+    'ST': NameOID.STATE_OR_PROVINCE_NAME,
+    'O': NameOID.ORGANIZATION_NAME,
+    'OU': NameOID.ORGANIZATIONAL_UNIT_NAME,
+    'L': NameOID.LOCALITY_NAME,
+    'CN': NameOID.COMMON_NAME,
+    'EMAIL': NameOID.EMAIL_ADDRESS,
+}
+
+
+def _create_csr(subject: str) -> tuple[str, str]:
+    logger.info(f'Creating CSR for {subject}')
+    key = rsa.generate_private_key(public_exponent=65537, key_size=2048)
+    key_pem = key.private_bytes(
+        encoding=serialization.Encoding.PEM,
+        format=serialization.PrivateFormat.TraditionalOpenSSL,
+        encryption_algorithm=serialization.NoEncryption(),
+    ).decode('utf-8')
+    attributes = []
+    for part in subject.split(', '):
+        (name, value) = part.split('=')
+        oid = _str_to_names[name]
+        attributes.append(x509.NameAttribute(oid, value))
+    csr = x509.CertificateSigningRequestBuilder().subject_name(
+        x509.Name(attributes)
+    ).sign(key, hashes.SHA256())
+    csr_pem = csr.public_bytes(serialization.Encoding.PEM).decode('utf-8')
+    return csr_pem, key_pem
+
+
+def _extract_client_certificate(cert_response: dict) -> str:
+    pkcs7_pem = cert_response['value']
+    certs = pkcs7.load_pem_pkcs7_certificates(pkcs7_pem.encode('utf-8'))
+    if len(certs) == 0:
```

Comment:
nit: it is considered to be more "pythonic" to use falseness of empty collections, like so:

```
if not certs:
  raise VE..
```